### ISSUE TITLE
Warnings in Unity console

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#415]
<sup>Issue created by Leslie Young as [Bitbucket Issue #&#8203;276](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276) on 2019.02.26 06:54.</sup>
This probably carry no priority at all but would be nice if the warnings in the Unity console could be sorted out. Personally I do not want to see any warnings and errors in the console after compiling code.

I grabbed latest version of repo and there is a warning about UIParticleSystem.OnDestroy() hiding UIBehaviour.OnDestroy() for example.

There are also a bunch of the "CS0649: Field '......' is never assigned to" which are easy to fix by just adding some default value or a pragma in the relevant script.

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.02.26 12:38](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50787484), @SimonDarksideJ commented:</sup>
Interesting, there shouldn't be any such errors and I work painlessly to ensure it stays that way, so a little surprised to see you found some.
What version of Unity are you currently using the UI Extensions in @plyoung ?

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.02.26 13:04](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50788087), Leslie Young commented:</sup>
Unity 2018.3.6f1.

CS0649 will normally not come up in the Unity console again if the relevant file is not touched. A re-import of the "unity-ui-extensions" folder can make those warnings appear again in the Unity console.

A rebuild of the solution in Visual Studio will also show them in the Error List. They are not critical at all though,

Just that OnDestroy override might need a fix.

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.02.27 14:43](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50809649), David Tso commented:</sup>
Same thing happened to me as well. About 32 warnings appear every time I edit my scripts. Is there any way to fix them? my Unity version is 2018.3.0f2

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.02.27 14:56](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50809942), Leslie Young commented:</sup>
David, add a value to the fields which causes the warning and it will go away. For example, in `ScrollPositionController.cs` the `viewport` field causes warning. Just make it " `viewport=null`. It willl happen on any private field marked with `[SerializeField]` that does not have a value assigned (where you would expect the default value to be used).

You can also add `#pragma warning disable 0649` at top, or just before the field list, of every script that causes this.

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.02.27 15:37](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50811025), David Tso commented:</sup>
Wow It worked! Thank you so mush :)) I didn't know that I can disable warning by script

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.03.05 15:46](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50905157), @SimonDarksideJ commented:</sup>
OK, picking this up today and hope to have it updated in source soon

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.03.07 17:26](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50945968), @SimonDarksideJ commented:</sup>
FIxed the onDestroy and going to release a patch release to the downloadable asset.

Tested in Both 2018.2 and 2018.3 (even 2019.1) and I can't replicate the  CS649 errors appearing in Unity, only in VS (because VS doesn't always understand Unity)  But I'll see if I can find a better solution than using Pragma's as they aren't nice.  Fixing is better than hiding :D

**Comment by SimonDarksideJ** [comment]
<sup>On [2019.03.07 18:15](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/276#comment-50946644), @SimonDarksideJ commented:</sup>
Right, spent a few hours combing through source and in builds and I believe I've caught all the warnings in this build.  We did a screening a while back, but some of the updates / newer controls hadn't been as thoroughly cleaned.  All sorted now.
Will do a patch release once I've finished my rounds of testing. (multiplied now thanks to the prefab changes in 2018.3 and the revamped Unity in 2019.1)
